src/fadc.jl

```diff
@@ -75,13 +75,26 @@ Simulate effects of FADC module <fadc> on the waveform <wf>
 """
 function simulate(wf::RDWaveform, fadc::GenericFADC)   
     T = Float32 # This should be somehow defined and be passed properly
-    # resample -> currently simply pick every Nth sample, more elaborate simulation later
+    ## resample
+    
+    # v0: simply pick every Nth sample of given sampling interval
     wf_sampled = wf.signal[begin : Int(fadc.Δt/step(wf.time)) : end]
+    
+    # v1: take mean of bins (sampling intervals) to smooth out noise
+    # -> actually much worse, results in too smooth and sharp FWHM etc.
+    # probably not how actual FADC works either
+    # wf_sampled = Vector{T}([])
+    # stp = Int(fadc.Δt/step(wf.time))
+    # for idx in 1:stp:length(wf.time)
+    #     last = idx+stp > length(wf.time) ? length(wf.time) : idx+stp
+    #     push!(wf_sampled, mean(wf.signal[idx:last]))
+    # end
+
     t_sampled = range(T(0)u"ns", step = fadc.Δt, length = length(wf_sampled))
     wf_daq = RDWaveform(t_sampled, wf_sampled)
 
     # digitize
     # wf_daq = RDWaveform(wf_daq.time, UInt16.(round.(wf_daq.signal, digits = 0)))
     wf_daq = RDWaveform( wf_daq.time, UInt16.(clamp.(round.(wf_daq.signal, digits = 0), 0, typemax(UInt16))) )
```

Comment:
This looks to me like it could be done in one line without the need to create the intermediate waveform object

Reply:
You mean do sampling and digitizing in one step?

Reply:
Exactly, but we can also leave this as is